# Decision making for opening a restaurant

### New York Historical  Population - Wikipedia

In [604]:
newyork_df=pd.DataFrame(newyork_A, columns=['Year'])
newyork_df['New York Population']=newyork_B
newyork_df

Year  New York Population
0   1698                 4937
1   1712                 5840
2   1723                 7248
3   1737                10664
4   1746                11717
5   1756                13046
6   1771                21863
7   1790                49401
8   1800                79216
9   1810               119734
10  1820               152056
11  1830               242278
12  1840               391114
13  1850               696115
14  1860              1174779
15  1870              1478103
16  1880              1911698
17  1890              2507414
18  1900              3437202
19  1910              4766883
20  1920              5620048
21  1930              6930446
22  1940              7454995
23  1950              7891957
24  1960              7781984
25  1970              7894862
26  1980              7071639
27  1990              7322564
28  2000              8008278
29  2010              8175133
30  2018              8398748

### Toronto Historical  Population - Wikipedia

In [603]:
toronto_df=pd.DataFrame(toronto_A, columns=['Year'])
toronto_df['Toronto Population']=toronto_B
toronto_df

Year  Toronto Population
0   1833                9252
1   1900              238080
2   1910              381383
3   1920              521893
4   1930              856955
5   1940              951549
6   1950             1176622
7   1960             1824481
8   1970             2089729
9   1975             2124291
10  1980             2137395
11  1985             2192721
12  1990             2275771
13  1995             2385421
14  2000             2481494
15  2005             2503281
16  2010             2615060
17  2015             2731571

### NewYork and Toronto merge

In [605]:
merge_dfs2 = merge_dfs1.fillna(method='ffill', limit=1) # fill values forward with limit
merge_dfs2

New York Population  Toronto Population
Year                                         
1900            3437202.0            238080.0
1910            4766883.0            381383.0
1920            5620048.0            521893.0
1930            6930446.0            856955.0
1940            7454995.0            951549.0
1950            7891957.0           1176622.0
1960            7781984.0           1824481.0
1970            7894862.0           2089729.0
1975            7894862.0           2124291.0
1980            7071639.0           2137395.0
1985            7071639.0           2192721.0
1990            7322564.0           2275771.0
1995            7322564.0           2385421.0
2000            8008278.0           2481494.0
2005            8008278.0           2503281.0
2010            8175133.0           2615060.0
2015            8175133.0           2731571.0
2018            8398748.0           2731571.0

### New York and Toronro comparsion

|    item     | New York           | Toronro  |
| ------------- |:-------------:| -----:|
|Population       | 8,175,133 | 2,954,024 |
| Population Density (/sq mi)      | 27,751      |   12,140 |
|GDP (US$ billion) | 842.3  (2018)   |   276.3  (2014) |

# 3. Choice for Borough in New York City

In [575]:
newyork_Boroughs_df=pd.DataFrame(newyork_Boroughs_A, columns=['Borough'])
newyork_Boroughs_df['County']=newyork_Boroughs_B
newyork_Boroughs_df['Population']=newyork_Boroughs_C
newyork_Boroughs_df['Gross Domestic Product (billons US$)']=newyork_Boroughs_D
newyork_Boroughs_df['Gross Domestic Product (per capita US$)']=newyork_Boroughs_E
newyork_Boroughs_df['Land area (square miles)']=newyork_Boroughs_F
newyork_Boroughs_df['Land area (square km)']=newyork_Boroughs_G
newyork_Boroughs_df['Density (persons / sq. mi)']=newyork_Boroughs_H
newyork_Boroughs_df['Density (persons / sq. km)']=newyork_Boroughs_I
newyork_Boroughs_df

Borough       County  Population  \
0          The Bronx        Bronx   1,432,132   
1           Brooklyn        Kings   2,582,830   
2          Manhattan     New York   1,628,701   
3             Queens       Queens   2,278,906   
4      Staten Island     Richmond     476,179   
5   City of New York                8,398,748   
6  State of New York               19,745,289   

  Gross Domestic Product (billons US$)  \
0                               42.695   
1                               91.559   
2                              600.244   
3                               93.310   
4                               14.514   
5                              842.343   
6                            1,701.399   

  Gross Domestic Product (per capita US$) Land area (square miles)  \
0                                  29,200                    42.10   
1                                  34,600                    70.82   
2                                 360,600                    22.83   
3                                  39,600                   108.53   
4                                  30,300                    58.37   
5                                  97,700                   302.64   
6                                  85,700                   47,214   

  Land area (square km) Density (persons / sq. mi) Density (persons / sq. km)  
0                109.04                     34,653                     13,231  
1                183.42                     37,137                     14,649  
2                 59.13                     72,033                     27,826  
3                281.09                     21,460                      8,354  
4                151.18                      8,112                      3,132  
5                783.83                     28,188                     10,947  
6               122,284                      416.4                        159

# 4. Search Neighborhoods in Manhattan

## Five downtown of Manhattan, New York city

## Neighborhoods Numbers of Five downtown in Manhattan

In [587]:
Manhattan_neighborhoods_df=pd.DataFrame(Manhattan_neighborhoods_A, columns=['Manhanttan downtown'])
Manhattan_neighborhoods_B = [Manhattan_neighborhoods_B1, Manhattan_neighborhoods_B2, Manhattan_neighborhoods_B3, Manhattan_neighborhoods_B4, Manhattan_neighborhoods_B5]
M_B1_len = len(Manhattan_neighborhoods_B1)
M_B2_len = len(Manhattan_neighborhoods_B2)
M_B3_len = len(Manhattan_neighborhoods_B3)
M_B4_len = len(Manhattan_neighborhoods_B4)
M_B5_len = len(Manhattan_neighborhoods_B5)
Manhattan_neighborhoods_B_len =[M_B1_len, M_B2_len, M_B3_len, M_B4_len, M_B5_len]
Manhattan_neighborhoods_df['Neighborhoods Numbers']=Manhattan_neighborhoods_B_len
Manhattan_neighborhoods_df['Neighborhoods']=Manhattan_neighborhoods_B
Manhattan_neighborhoods_df

Manhanttan downtown  Neighborhoods Numbers  \
0     Lower Manhattan                     25   
1             Midtown                     11   
2           West Side                     12   
3           East Side                     16   
4     Upper Manhattan                     14   

                                       Neighborhoods  
0  [Alphabet City, Battery Park City, Bowery, Chi...  
1  [Columbus Circle, Diamond District, Flatiron D...  
2  [Chelsea, Hell's Kitchen, Hudson Yards, develo...  
3  [Carnegie Hill, Gramercy Park, Kips Bay, Lenox...  
4  [Astor Row, East Harlem, Hamilton Heights, Har...

In [581]:
neighborhoods_df = Manhattan_neighborhoods_df['Neighborhoods'][1]

In [606]:
neighborhood_df = pd.DataFrame(neighborhoods_df, columns=['Midtown'])
neighborhood_df

Midtown
0     Columbus Circle
1    Diamond District
2   Flatiron District
3    Garment District
4       Herald Square
5           Koreatown
6      Madison Square
7               NoMad
8       Silicon Alley
9    Theater District
10       Times Square

## Venues Numbers in Neighborhoods by Foursquare API

In [590]:
neighborhood_df['latitude'] = latitude_list
neighborhood_df['longitude'] = longitude_list
neighborhood_df['Venue numbers'] = venue_len_list
neighborhood_df['Venue numbers'].sum()

1356

In [591]:
Manhattan_neighborhoods_df['Venue numbers'] = [2768, 1356, 1242, 1561, 1228]
Manhattan_neighborhoods_df['Venue per Neighborhoods'] = [111, 123, 104, 98, 88]
Manhattan_neighborhoods_df

Manhanttan downtown  Neighborhoods Numbers  \
0     Lower Manhattan                     25   
1             Midtown                     11   
2           West Side                     12   
3           East Side                     16   
4     Upper Manhattan                     14   

                                       Neighborhoods  Venue numbers  \
0  [Alphabet City, Battery Park City, Bowery, Chi...           2768   
1  [Columbus Circle, Diamond District, Flatiron D...           1356   
2  [Chelsea, Hell's Kitchen, Hudson Yards, develo...           1242   
3  [Carnegie Hill, Gramercy Park, Kips Bay, Lenox...           1561   
4  [Astor Row, East Harlem, Hamilton Heights, Har...           1228   

   Venue per Neighborhoods  
0                      111  
1                      123  
2                      104  
3                       98  
4                       88

# 5. Decision of Neighborhoods in Midtown -> Flatiron District

In [593]:
neighborhood_df['Venue numbers'] = venue_len_list
neighborhood_df

Midtown latitude longitude  Venue numbers
0     Columbus Circle  40.7681  -73.9819            118
1    Diamond District  40.7572  -73.9801            128
2   Flatiron District  40.7411  -73.9897            133
3    Garment District  40.7537  -73.9905            125
4       Herald Square  40.7495   -73.988            120
5           Koreatown  40.7476  -73.9865            128
6      Madison Square  42.8759  -75.6803            121
7               NoMad  40.7447  -73.9883            119
8       Silicon Alley                               119
9    Theater District   40.758  -73.9855            123
10       Times Square  40.7573  -73.9859            122

## Map of Neighborhoods in Midtown

In [497]:
address = '102 North End Ave, New York, NY'
address1 = 'Midtown, New York, NY'
geolocator = Nominatim(user_agent="foursquare_agent")
location = geolocator.geocode(address1)
latitude = location.latitude
longitude = location.longitude
print(latitude, longitude)

40.7622684 -73.9795443


In [499]:
venues_map = folium.Map(location=[latitude, longitude], zoom_start=13) # generate map centred around the Conrad Hotel

# add a red circle marker to represent the Conrad Hotel
folium.features.CircleMarker(
    [latitude, longitude],
    radius=10,
    color='red',
    popup='Midtown',
    fill = True,
    fill_color = 'red',
    fill_opacity = 0.6
).add_to(venues_map)

# add the Italian restaurants as blue circle markers
for lat, lng, label in zip(latitude_list, longitude_list, neighborhood_df.Midtown):
    folium.features.CircleMarker(
        [lat, lng],
        radius=5,
        color='blue',
        popup=label,
        fill = True,
        fill_color='blue',
        fill_opacity=0.6
    ).add_to(venues_map)

# display map
venues_map

# 6. Search Venues in Flatiron District by Foursquare API

In [595]:
CLIENT_ID = 'EM1ADUZZU3VQ2SYZ5QCZZJM4BQVZFBVGZ0C2LJ1QRK1RTFV4' # your Foursquare ID
CLIENT_SECRET = '5AH3XD0QGPDTT4WFFASFPCSEKWCFKASWTMDLZZBEO0ZYSHJX' # your Foursquare Secret
VERSION = '20180604'
LIMIT = 20

In [596]:
search_query = 'korean'
radius = 500

In [597]:
url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&query={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude, VERSION, search_query, radius, LIMIT)

In [598]:
results = requests.get(url).json()
#results

In [599]:
# assign relevant part of JSON to venues
venues = results['response']['venues']

# tranform venues into a dataframe
dataframe = json_normalize(venues)
print(dataframe.shape)
len(dataframe)
dataframe.head()

(4, 17)


id                                  name  \
0  4e15fc64b61c42e7c5506dcb  Korean American Community Foundation   
1  55491797498ee29add05e99c                 fire belly korean bbq   
2  5b7450cc82a750002cb63c1a          KumGangSan Korean Restaurant   
3  4ca2818bf88a95213f5fa149                             Soup Cart   

                                          categories    referralId  hasPerk  \
0  [{'id': '4bf58dd8d48988d124941735', 'name': 'O...  v-1583304948    False   
1  [{'id': '4bf58dd8d48988d1df931735', 'name': 'B...  v-1583304948    False   
2  [{'id': '4bf58dd8d48988d113941735', 'name': 'K...  v-1583304948    False   
3  [{'id': '4bf58dd8d48988d1cb941735', 'name': 'F...  v-1583304948    False   

  location.address location.crossStreet  location.lat  location.lng  \
0      501 5th Ave              42nd St     40.752644    -73.981663   
1              NaN                  NaN     40.754860    -73.986730   
2     16 W 48th St                  NaN     40.757530    -73.979326   
3       W 45th St.          at 6th Ave.     40.756901    -73.982369   

                             location.labeledLatLngs  location.distance  \
0  [{'label': 'display', 'lat': 40.75264420000000...                625   
1  [{'label': 'display', 'lat': 40.75485957453933...                279   
2  [{'label': 'display', 'lat': 40.75752999999999...                551   
3  [{'label': 'display', 'lat': 40.75690098944781...                296   

  location.postalCode location.cc location.city location.state  \
0               10017          US      New York             NY   
1                 NaN          US      New York             NY   
2               10036          US      New York             NY   
3               10036          US      New York             NY   

  location.country                          location.formattedAddress  
0    United States  [501 5th Ave (42nd St), New York, NY 10017, Un...  
1    United States                      [New York, NY, United States]  
2    United States  [16 W 48th St, New York, NY 10036, United States]  
3    United States  [W 45th St. (at 6th Ave.), New York, NY 10036,...

In [541]:
# keep only columns that include venue name, and anything that is associated with location
filtered_columns = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
dataframe_filtered = dataframe.loc[:, filtered_columns]
#dataframe_filtered
# clean column names by keeping only last term
dataframe_filtered.columns = [column.split('.')[-1] for column in dataframe_filtered.columns]
dd = dataframe_filtered.columns
dd

Index(['name', 'categories', 'address', 'crossStreet', 'lat', 'lng',
       'labeledLatLngs', 'distance', 'postalCode', 'cc', 'city', 'state',
       'country', 'formattedAddress', 'id'],
      dtype='object')

In [542]:
# keep only columns that include venue name, and anything that is associated with location
filtered_columns = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
dataframe_filtered = dataframe.loc[:, filtered_columns]
dd = dataframe_filtered
# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

# filter the category for each row
dataframe_filtered['categories'] = dataframe_filtered.apply(get_category_type, axis=1)

# clean column names by keeping only last term
dataframe_filtered.columns = [column.split('.')[-1] for column in dataframe_filtered.columns]

dataframe_filtered

name         categories       address  \
0                     Korean Pro  Korean Restaurant  62 W 56th St   
1             Korean Air Offices             Office  10 E 53rd St   
2                  Korean Market  Korean Restaurant           NaN   
3   KumGangSan Korean Restaurant  Korean Restaurant  16 W 48th St   
4             Korean Air Offices             Office   609 5th Ave   
5  B O B Korean Spicy Restaurant  Korean Restaurant            68   

   crossStreet        lat        lng  \
0  at 6th Ave.  40.763404 -73.977167   
1      5th Ave  40.759687 -73.975566   
2          NaN  40.763979 -73.985287   
3          NaN  40.757530 -73.979326   
4   W 50th St.  40.757624 -73.977526   
5          NaN  40.766729 -73.982621   

                                      labeledLatLngs  distance postalCode  cc  \
0  [{'label': 'display', 'lat': 40.76340383711987...       236      10019  US   
1  [{'label': 'display', 'lat': 40.759687230922, ...       441      10022  US   
2  [{'label': 'display', 'lat': 40.763979, 'lng':...       520        NaN  US   
3  [{'label': 'display', 'lat': 40.75752999999999...       527      10036  US   
4  [{'label': 'display', 'lat': 40.75762433612029...       544      10017  US   
5  [{'label': 'display', 'lat': 40.76672878222091...       560      10110  US   

       city state        country  \
0  New York    NY  United States   
1  New York    NY  United States   
2  New York    NY  United States   
3  New York    NY  United States   
4  New York    NY  United States   
5     Asoke    NY  United States   

                                    formattedAddress                        id  
0  [62 W 56th St (at 6th Ave.), New York, NY 1001...  4a52ac4ef964a520e5b11fe3  
1  [10 E 53rd St (5th Ave), New York, NY 10022, U...  5a96b1a0d4cc987c4a7344dc  
2                      [New York, NY, United States]  51c475aa498e73948caf7ebd  
3  [16 W 48th St, New York, NY 10036, United States]  5b7450cc82a750002cb63c1a  
4  [609 5th Ave (W 50th St.), New York, NY 10017,...  4f076211e4b090ef20d24084  
5               [68, Asoke, NY 10110, United States]  4c7c9b689efda143f64a86e2

## Map of Korean food in Midtown

In [543]:
venues_map = folium.Map(location=[latitude, longitude], zoom_start=13) # generate map centred around the Conrad Hotel

# add a red circle marker to represent the Conrad Hotel
folium.features.CircleMarker(
    [latitude, longitude],
    radius=10,
    color='red',
    popup='Midtown',
    fill = True,
    fill_color = 'red',
    fill_opacity = 0.6
).add_to(venues_map)

# add the Italian restaurants as blue circle markers
for lat, lng, label in zip(dataframe_filtered.lat, dataframe_filtered.lng, dataframe_filtered.name):
    folium.features.CircleMarker(
        [lat, lng],
        radius=5,
        color='blue',
        popup=label,
        fill = True,
        fill_color='blue',
        fill_opacity=0.6
    ).add_to(venues_map)

# display map
venues_map